### PR TITLE
doc: update Go snippets and remove Beta status

### PR DESCRIPTION
**Pull request by falconandy**
# Amplitude Developer Docs PR

## Description

Go: updated examples, removed Beta status.

Added more Go configuration options - descriptions are copied from Python. 
Not added: configuration parameter `FlushSizeDivider` - I don't know how to describe it properly. Maybe it should be private/internal.
Markdown table with configuration parameters was reformatted by IDE. I hope it shouldn't break anything.

## Deadline

When https://github.com/amplitude/analytics-go will be released to `v1`

## Change type

- [x] Doc bug fix. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Comment by justin-fiedler** [comment]
Thanks @falconandy . No need to include FlushSizeDivider in the docs. I think that could be made internal.

File: docs/data/sdks/go/ampli.md [hunk 1]
```diff
@@ -106,9 +107,8 @@ ampli.Instance.Load(ampli.LoadOptions{
 
 | <div class ="big-column">Arg of LoadOptions</div> | Description |
 |-|-|
-|`Environment`| Required. String. Specifies the environment the Ampli Wrapper is running in. For example,  `EnvironmentProduction` or `EnvironmentDevelopment`. Create, rename, and manage environments in Amplitude Data.<br /><br />Environment determines which API token is used when sending events.<br /><br />If a `Client.ApiKey` or `Client.Instance` is provided, `Environment` is ignored, and can be omitted.|
+|`Environment`| Required. String. Specifies the environment the Ampli Wrapper is running in. For example,  `EnvironmentProd` or `EnvironmentDev`. Create, rename, and manage environments in Amplitude Data.<br /><br />Environment determines which API token is used when sending events.<br /><br />If a `Client.ApiKey` or `Client.Instance` is provided, `Environment` is ignored, and can be omitted.|
```
Comment (justin-fiedler) [inline docs/data/sdks/go/ampli.md:110]: I need to double check on the default names, but I think this should be: `EnvironmentProduction` or `EnvironmentDevelopment`

**Review by justin-fiedler — APPROVED** [review]

File: docs/data/sdks/go/ampli.md (hunk 1 above)
Comment (falconandy) [reply docs/data/sdks/go/ampli.md]: Fixed

Comment (justin-fiedler) [reply docs/data/sdks/go/ampli.md]: Thanks @falconandy. I confirmed with the Data team 

> it currently depends if it’s a migrated project or a not - production for new projects

**Pull request merged**